Spine/ResourceField.swift

```diff
@@ -7,98 +7,614 @@
+
+    public init(_ name: String, to linkedType: T.Type) {
+        self.name = name
+        self.serializedName = name
+    }
+
+    public func serialize(from resource: Resource,
+                          into serializedData: inout [String: Any],
+                          withKeyFormatter keyFormatter: KeyFormatter,
+                          withValueFormatters valueFormatters: ValueFormatterRegistry,
+                          withOptions options: SerializationOptions) {
+        let key = keyFormatter.format(self)
+
+        Spine.logDebug(.serializing, "Serializing toOne relationship \(self) as '\(key)'")
+
+        if options.contains(.IncludeToOne) {
+            let serializedId: Any
+            let linkedResource = resource.value(forField: self.name) as? Linked
+
+            if let resourceId = linkedResource?.id {
+                serializedId = resourceId
+            } else {
+                serializedId = NSNull()
+            }
+
+            let serializedRelationship = [
+                "data": [
+                    "type": Linked.resourceType,
+                    "id": serializedId
+                ]
+            ]
+
+            if serializedData["relationships"] == nil {
+                serializedData["relationships"] = [key: serializedRelationship]
+            } else {
+                var relationships = serializedData["relationships"] as! [String: Any]
+                relationships[key] = serializedRelationship
+                serializedData["relationships"] = relationships
+            }
+        }
+
+    }
+
+    public func extract(from serializedData: JSON,
+                        intoResource resource: Resource,
+                        withKeyFormatter keyFormatter: KeyFormatter,
+                        withValueFormatters valueFormatters: ValueFormatterRegistry,
+                        fromResourcePool pool: inout [Resource],
+                        withFactory factory: ResourceFactory) {
+        let key = keyFormatter.format(self)
+        resource.relationships[self.name] = self.extractRelationshipData(serializedData["relationships"][key])
+
+        var linkedResource: T? = nil
+
+        if let linkData = serializedData["relationships"][key].dictionary {
+            // XXX: assert(linkData["data"]?["type"].string == Linked.resourceType)
+
+            if let id = linkData["data"]?["id"].string {
+                linkedResource = factory.dispense(Linked.self, id: id, pool: &pool)
+            } else {
+                linkedResource = Linked.init()
+            }
+
+            if let resourceURL = linkData["links"]?["related"].url {
+                linkedResource!.url = resourceURL
+            }
+        }
+        
+        if let linkedResource = linkedResource {
+            if resource.value(forField: self.name) == nil || (resource.value(forField: self.name) as? Resource)?.isLoaded == false {
+                resource.setValue(linkedResource, forField: self.name)
```

Comment:
@mathebox: by the looks of it ToOneRelationship shouldn't be resolving the relationship here since the deserialise operation hasn't passed included yet...
https://github.com/mathebox/Spine/blob/741a0943fdbfd48ee149603eda1effe8678baa54/Spine/DeserializeOperation.swift#L82